Automattic/jetpack
# Tiled Gallery: Fix block validation errors caused by inconsistent flex-basis style

Fixes #16514 
Closes #16543 

A change made to the Tiled Gallery last year to fix mosaic and column layouts on AMP pages (https://github.com/Automattic/jetpack/pull/15770) introduced a block validation issue. This was a result of applying a decimal percentage width to columns via an inline flex-basis style. In the actual block content those decimal values were rounded to 12 places whereas the save function allowed 15. This inconsistency meant the two didn't match and caused the error.

This PR is essentially a clone of https://github.com/Automattic/jetpack/pull/16543 with fixes applied so the v3 deprecation runs correctly. I ran into issues rebasing the original PR after the recent Jetpack restructure hence this fresh PR.
	
#### Changes proposed in this Pull Request:
* Restrict the percentage columns widths applied as flex-basis inline style to 5 decimal places
* Add deprecation to migrate blocks with a flex-basis style already applied

#### Does this pull request change what data or activity we track or use?
No change.

#### Testing instructions:
* On a wpcom site, create a new post, insert a layout grid and tiled galleries within its columns
* Assign one gallery the mosaic block style and another the columns style
* Save the post and reload the editor, you should receive block validation errors
* Sync this PR or apply its diff to your sandbox and sandbox your test site
* Reload the editor again and you should not encounter the block validation error
* In dev tools check the console logs to ensure the deprecation ran and the block successfully migrated 

#### Screenshot
<img width="2441" alt="Screen Shot 2021-03-02 at 11 49 52 pm" src="https://user-images.githubusercontent.com/60436221/109661922-377aef80-7bb6-11eb-849c-4fa643865d50.png">

#### Proposed changelog entry for your changes:
* Bug Fix: Prevent block validation errors for tiled galleries with mosaic or column layouts.
